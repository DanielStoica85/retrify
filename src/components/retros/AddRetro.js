import React from 'react';
import { connect } from 'react-redux';
import { addRetro } from '../../actions/retros';
import { Container } from 'reactstrap';
import RetroForm from './RetroForm';

export class AddRetro extends React.Component {
    addRetro = retro => {
        this.props.addRetro(retro);
        this.props.history.push('/');
    };

    render() {
        return (
            <Container>
                <h4 className="grey-text text-darken-3">Add Retro Board</h4>
                <RetroForm onSubmit={this.addRetro} />
            </Container>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        addRetro: retro => dispatch(addRetro(retro))
    };
};

export default connect(
    undefined,
    mapDispatchToProps
)(AddRetro);
