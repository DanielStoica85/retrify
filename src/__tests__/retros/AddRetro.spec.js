import React from 'react';
import renderer from 'react-test-renderer';
import AddRetro from '../../components/retros/AddRetro';

describe('AddRetro component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<AddRetro />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
