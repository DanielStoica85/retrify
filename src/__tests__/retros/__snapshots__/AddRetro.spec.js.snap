// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRetro component renders correctly 1`] = `
<div
  className="container"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <h4
      className="grey-text text-darken-3"
    >
      Add Retro Board
    </h4>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="title"
      >
        Retro Board Name
      </label>
      <input
        className="form-control"
        id="title"
        name="title"
        onChange={[Function]}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="description"
      >
        Description
      </label>
      <textarea
        className="form-control"
        id="description"
        name="description"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-group"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        onClick={[Function]}
      >
        Add Retro
      </button>
    </div>
  </form>
</div>
`;
